CDCgov/prime-reportstream
# Lpabon/6178 delivery details data

This PR connects the Delivery Details endpoint to the database so that actual data may be pulled from it.

Test Steps:
1. Submit a report on your dev environment. You can do so via the CLI with the following: `curl -X POST -H "client:simple_report" -H "Content-Type: text/csv" --data-binary "@./examples/submission/simple_report_example.csv" "http://localhost:7071/api/reports"`
2. Wait a few minutes while the test reports are being sent, then make a GET query to `/api/waters/org/ak-phd.elr/deliveries/`. Write down the deliveryId and reportId values for one of the results. If `ak-phd.elr` does not contain any deliveries, use the Submissions API to find a Destination with new reports.
3. Make a GET query to `/api/waters/report/{id}/delivery` using the `deliveryId` value from the previous step for the `id`. Compare the results with the schema defined in `docs/api/waters-delivery.yml`
4. Make a GET query to `/api/waters/report/{id}/facilities` using the `reportId` value from step 2 for the `id`. Verify the returned list is identical to the one returned in the previous step.

## Changes
- Updated the data logic for Delivery Details from mock data to actual DB queries.

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
